src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -242,6 +244,19 @@ public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<T> source)
             return source.Where((Func<T, bool>)s_notNullTest);
         }
 
+        public static ImmutableArray<TResult> SelectAsArray<TSource, TResult>(this IEnumerable<TSource> source, Func<TSource, TResult> selector)
```

Comment:
💭 I thought we already had this extension method. Maybe not accessible here though?